src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -280,12 +281,7 @@ private boolean isDiscarded() {
          *         associated with it. true: Reference was successfully added.
          */
         boolean tryAddReference() {
-            if (isDiscarded() || isExplicitlyDiscarded())
-                return false;
-            else {
-                int refCount = handleRefCount.incrementAndGet();
-                return refCount > 0;
-            }
+            return isDiscarded() || isExplicitlyDiscarded() ? false : handleRefCount.incrementAndGet() > 0;
```

Comment:
I think I mentioned similar comments before.. please make sure conditions are always in brackets if there's OR or AND clause in the condition - so the interpreter does not get confused.
```suggestion
            return (isDiscarded() || isExplicitlyDiscarded()) ? false : handleRefCount.incrementAndGet() > 0;
```